gadgets/trace_tcp/ebpf/close.h

```diff
@@ -26,19 +43,35 @@ static __always_inline void handle_tcp_close(struct pt_regs *ctx,
 	u8 oldstate = BPF_CORE_READ(sk, __sk_common.skc_state);
 	if (oldstate == TCP_SYN_SENT || oldstate == TCP_SYN_RECV ||
 	    oldstate == TCP_NEW_SYN_RECV)
-		return;
+		goto end;
 
 	family = BPF_CORE_READ(sk, __sk_common.skc_family);
 	if (!fill_tuple(&tuple, sk, family))
-		return;
+		goto end;
 
 	event = gadget_reserve_buf(&events, sizeof(*event));
 	if (!event)
-		return;
+		goto end;
 
 	fill_event(event, &tuple, NULL, 0, close);
```

Comment:
This fills the event with the current process. But the fd below might or might not refer to the current process, depending if we get it from connect/accept time or from the close syscall.

This is confusing to document, so I suggest to drop the fd for close events at the moment. I have heard requests for fd in the accept events but not for the close events.

Example of sequence:
```
fd1 = socket();
fd2 = dup(fd);
fd3 = dup(fd);
ret = connect(fd2, localhost:8080); // -> generate connect event
close(fd3);
close(fd2);
close(fd1);  // -> generate close event
```